TSG/Update/cau-debug-in-23h2.md

```diff
@@ -0,0 +1,263 @@
+> ...
+> [06/26/2024 08:17:09] [16]: Canceling any remaining retries.
+> [06/26/2024 08:17:09] [16]: WriteError1: (no details)
+> [06/26/2024 08:17:09] [16]: WriteError2: InvalidOperation: `(:)` [], ClusterUpdateException
+> [06/26/2024 08:17:09] [16]: WriteError3: 
+<mark>PluginReported_InstallFailure_AudiInstall_PayloadNotFound_0x80040406</mark>
+> [06/26/2024 08:17:09] [16]: WriteError4: ClusterUpdateException: The Microsoft.RollingUpgradePlugin plug-in reported a failure while attempting to install updates on node "V-HOST3". Additional information reported by the plug-in: (ClusterUpdateException) Failed to start upgrade process because payload was not found with properties: "WuConnected: "False"; PathToSetupMedia: "Some\path";", on node: "V-HOST3". HRESULT: 0x80040406 ---> ClusterUpdateException: Failed to start upgrade process because payload was not found with properties: "WuConnected: "False"; PathToSetupMedia: "Some\path";", on node: "V-HOST3". HRESULT: 0x80040406
+
+### CAU wmi logs: 
+
+These logs trace the CAU wmi interactions with the update API, such as the detection, expansion, and installation of updates. The default location of these logs is <mark>**`%windir%\System32\LogFiles\ClusterUpdate\`**</mark> on each cluster node.
+
+> [06/26/2024 08:17:09 1b74] CauWmiV2!CauPersist::Save: Saved a CauNode for Guid= <mark>43703982-acaf-496d-bf20-97f859219631</mark>
+> [06/26/2024 08:17:09 1b74] +++++ CauRunDeployApiInstallOperation::CauRunDeployApiInstallOperation 
+> ...
+> [06/26/2024 08:17:09 1b74] CauWmiV2!AudiInterfaceHelper::RunUpgradeInstallUsingDeployApi: <mark>Failed to open payload file Some\path. Error 3</mark>
+> [06/26/2024 08:17:09 1b74] CauWmiV2!AudiInterfaceHelper::RunUpgradeInstallUsingDeployApi: Exiting
+> ...
+> [06/26/2024 08:17:09 1b74] CauWmiV2!CauRunDeployApiInstallOperation::RunUpgradeInstallUsingDeployApi: Audi Install CauOperation failed. Setup path: Some\path. Exit code: -2147220474.
+> ...
+> [06/26/2024 08:17:09 1b74] CauWmiV2!CauNode::RunUpgradeInstallAudiInterfaceImpl: Guid= 43703982-acaf-496d-bf20-97f859219631 failed MiResult== 0x00000000 HRESULT= <mark>0x80040406</mark>
+> [06/26/2024 08:17:09 1b74] CauWmiV2!CauNode::RunUpgradeInstallAudiInterfaceImpl: Audi Installer run completed. Exit code: -2147220474.
+
+### Cluster logs:
+
+These logs may be helpful to dive deeper into the cluster infrastructure issues, such as the cluster communication, the quorum, the role state, and the node drain and resume status. The default location of these logs is <mark>**`%windir%\Cluster\Reports\Cluster.log`**</mark> on each cluster node, can also be collected using [Get-ClusterLog (FailoverClusters) | Microsoft Learn](https://learn.microsoft.com/en-us/powershell/module/failoverclusters/get-clusterlog?view=windowsserver2022-ps) cmdlet.
+
+## Log analysis
+
+To diagnose and investigate problems, there are two possible scenarios:
+
+### Kusto
```

Comment:
Please remove references to internal Kusto queries since this TSG is meant for customer self-remediation.